src/eldap_filter_yecc.yrl

```diff
@@ -54,7 +54,7 @@ Erlang code.
 
 'and'(Value)                -> eldap:'and'(eldap_utils:maybe_b2list(Value)).
 'or'(Value)                 -> eldap:'or'(eldap_utils:maybe_b2list(Value)).
-'not'(Value)                -> eldap:'not'(eldap_utils:maybe_b2list(Value)).
+'not'(Value)                -> eldap:'not'(Value).
```

Comment:
I don't get it. `eldap_utils:maybe_b2list` was used consistently, and the change makes it inconsistent. Could you shed some light on why only the `not` operator is changed in this PR? Is this somehow justified, i.e. do you know why dialyzer only complained about this line?

Reply:
`and` and `or` takes a list of filters:

```
eldap:'and'([Filter1, Filter2, Filter3...]).
eldap:'or'([Filter1, Filter2, Filter3...]).
```

`not` takes a single filter:

```
eldap:'or'(Filter).
```

Reply:
Dialyzer errors:

```erlang
src/eldap_filter_yecc.yrl
Line 57 Column 1: Function 'not'/1 has no local return
Line 57 Column 44: The call eldap:'not'(maybe_improper_list() | {'error',{'unknown_type',_}}) breaks the contract (Filter) -> filter() when Filter :: {filter()}

src/eldap_utils.erl
Line 194 Column 43: The call eldap:equalityMatch(Name::any(), Value::binary()) breaks the contract (Type, Value) -> filter() when Type :: string(), Value :: string()

src/mongoose_ldap_worker.erl
Line 75 Column 1: Function connect/1 has no local return
Line 109 Column 1: Function retry_call_eldap/2 will never be called
===> Warnings written to _build/default/27.0.dialyzer_warnings
```